### ISSUE TITLE
I'm getting a platform exception when trying to use the client.search on the web

### ISSUE THREAD

**Issue by lifehug** [isoos/elastic_client#20]
Hi, the client works great for my use case on a device but when I try and call search from the web I get the following error. 

```
errors.dart:165 Uncaught (in promise) Error: Unsupported operation: Platform._version
    at Object.throw_ [as throw] (errors.dart:214)
    at Function._version (io_patch.dart:289)
    at Function.get version [as version] (platform_impl.dart:119)
    at get _version (platform.dart:74)
    at Function.desc.get [as _version] (operations.dart:781)
    at Function.get version [as version] (platform.dart:231)
    at Object._getHttpVersion (http_impl.dart:3374)
    at new _http._HttpClient.new (http_impl.dart:2157)
    at Function.new (http.dart:1556)
    at Function.new (console.dart:45)
    at new console_http_transport.ConsoleHttpTransport.new (console_http_transport.dart:10)
    at new elastic.Elastic.new (elastic.dart:13)
    at chat_screen._ChatScreenState.new.filteredStream (chat_screen.dart:81)
    at chat_screen.dart:66
    at chat_screen._ChatScreenState.new.setState (framework.dart:1236)
    at chat_screen._ChatScreenState.new.<anonymous> (chat_screen.dart:65)
    at Generator.next (<anonymous>)
    at runBody (async_patch.dart:84)
    at Object._async [as async] (async_patch.dart:123)
    at editable_text.EditableText.new.<anonymous> (chat_screen.dart:63)
    at editable_text.EditableTextState.new.[_formatAndSetValue] (editable_text.dart:1788)
    at editable_text.EditableTextState.new.updateEditingValue (editable_text.dart:1349)
    at text_input.TextInput.__._handleTextInputInvocation (text_input.dart:1121)
    at _handleTextInputInvocation.next (<anonymous>)
    at runBody (async_patch.dart:84)
    at Object._async [as async] (async_patch.dart:123)
    at text_input.TextInput.__.[_handleTextInputInvocation] (text_input.dart:1081)
    at OptionalMethodChannel._handleAsMethodCall (platform_channel.dart:430)
    at _handleAsMethodCall.next (<anonymous>)
    at runBody (async_patch.dart:84)
    at Object._async [as async] (async_patch.dart:123)
    at OptionalMethodChannel.[_handleAsMethodCall] (platform_channel.dart:427)
    at platform_channel.dart:383
    at handlePlatformMessage (binding.dart:265)
    at handlePlatformMessage.next (<anonymous>)
    at runBody (async_patch.dart:84)
    at Object._async [as async] (async_patch.dart:123)
    at _DefaultBinaryMessenger.handlePlatformMessage (binding.dart:256)
    at Object._invoke3 (window.dart:756)
    at _engine.EngineWindow.new.invokeOnPlatformMessage (window.dart:424)
    at _engine.TextEditingChannel.new.updateEditingState (text_editing.dart:1094)
    at _engine.GloballyPositionedTextEditingStrategy.new.<anonymous> (text_editing.dart:1280)
    at _engine.GloballyPositionedTextEditingStrategy.new.[_handleChange] (text_editing.dart:698)
    at Object._checkAndCall (operations.dart:324)
    at Object.dcall (operations.dart:329)
    at HTMLInputElement.<anonymous> (html_dart2js.dart:37175)
```

If helpful here is the code I'm using

```
class Elastic {
  static const URL =
      "URL";
  static const USERNAME = "USERNAME";
  static const PASSWORD = "PASSWORD";
  Transport transport;
  Client client;
  Elastic() {
    transport = ConsoleHttpTransport(Uri.parse(URL),
        basicAuth: BasicAuth(USERNAME, PASSWORD));
    client = elastic.Client(transport);
  }

  Future<List<MessageBubble>> search({
    String searchTerm,
    String senderEmail,
  }) async {
    List<MessageBubble> res;
    try {
      final SearchResult result = await client.search('messages', '_doc',
          elastic.Query.match('text', searchTerm, minimum: senderEmail),
          source: true);

      if (result.toMap()['totalCount'] == 0) {
        return [];
      } else {
        res = result
            .toMap()['hits']
            .map((e) => e['doc'])
            .map<MessageBubble>((e) => MessageBubble(
                text: e['text'],
                sender: e['sender'],
                isMe: senderEmail == e['sender']))
            .toList();
      }
    } catch (e) {
      print("this is a problem in web $e");
    }

    return res;
  }
}
```

**Comment by isoos** [comment]
@lifehug the core reason is that the underlying `package:http_client` is for console apps and not web apps (see the name `ConsoleHttpTransport`).

For you web app, you could rewrite this file: https://github.com/isoos/elastic_client/blob/master/lib/console_http_transport.dart

Instead of `import 'package:http_client/console.dart';` use `import 'package:http_client/browser.dart';`, and `ConsoleClient` could be `BrowserClient`.

In the mid-term I'm planning to migrate this to use `package:http`, which "transparently" uses the appropriate dependency based on the runtime.

**Comment by lifehug** [comment]
Hey, thanks for the help on this.  I wrote this myself with http today and ran into a weird issue that might save you some time when you get to it. Took a day of mine, so I thought I'd share. 

Elastic search returned different results when using GET with a body from the web than it does from a console. Hope I saved you some time. 

So don't use this https://www.elastic.co/guide/en/elasticsearch/reference/current/run-a-search.html#run-request-body-search

But use this https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-index.html
